### PR TITLE
docs: DBA, Database and Developer - The Decameron

### PR DESCRIPTION
**Pull request by LucyLu8L**
I‘ve added the md document and a static picture, please review.

### CODE REVIEW

File: content/blog/ddd-decameron.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: "DBA, Database and Developer - The Decameron"
+author: Lucy
+published_at: 2022/06/06 15:30
+feature_image: /static/blog-changelog-assets/2022/06/roaringdba.jpeg
+tags: Education
+featured: true
+description: This is the first of the DDD series, consisting of 10 stories of DBA & Database & developer.
+---
+# DBA, Database and Developer - The Decameron
```
Comment (boojack) [inline content/blog/ddd-decameron.md:10]: Don't need to add this line. We will render the header title according to `title: xxxx`.

File: content/blog/ddd-decameron.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+A few days ago, we posted a question on HN:[ Ask HN: Is DBA still a good job?](https://news.ycombinator.com/item?id=31309729) The question quickly drew the attention of the HN community and got 236 thought-provoking comments. We share and feel the challenges, the frustration, the fatigue and the expectation.
```
Comment (boojack) [inline content/blog/ddd-decameron.md:12]:
```suggestion
A few days ago, we posted a question on HN: [Ask HN: Is DBA still a good job?](https://news.ycombinator.com/item?id=31309729) The question quickly drew the attention of the HN community and got 236 thought-provoking comments. We share and feel the challenges, the frustration, the fatigue and the expectation. 
```

File: content/blog/ddd-decameron.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+A few days ago, we posted a question on HN:[ Ask HN: Is DBA still a good job?](https://news.ycombinator.com/item?id=31309729) The question quickly drew the attention of the HN community and got 236 thought-provoking comments. We share and feel the challenges, the frustration, the fatigue and the expectation. 
+
+We want to preserve all these heated discussions and help more people develop a multi-dimensional perspective on DBAs. That's how an idea for a blog series emerges -- **D·D·D** (DBA, Database, and Developer). 
+
+As the debut of this series, we compiled 10 stories about DBAs based on the replies and comments from the[ HN thread](https://news.ycombinator.com/item?id=31309729): we adopted the Decameron style, made up the story background, and borrowed the true stories shared by the commenters and turned them over to the storytellers (all credits go to the rightful owners). Some of these storytellers had experience as full-time DBAs; the others may be developers who have worked with DBAs before. You can regard this article as the appetizer of a full course meal. The crew welcomes all kinds of feedback and, last but not least: Bon Appetite!
```
Comment (boojack) [inline content/blog/ddd-decameron.md:16]:
```suggestion
As the debut of this series, we compiled 10 stories about DBAs based on the replies and comments from the [HN thread](https://news.ycombinator.com/item?id=31309729): we adopted the Decameron style, made up the story background, and borrowed the true stories shared by the commenters and turned them over to the storytellers (all credits go to the rightful owners). Some of these storytellers had experience as full-time DBAs; the others may be developers who have worked with DBAs before. You can regard this article as the appetizer of a full course meal. The crew welcomes all kinds of feedback and, last but not least: Bon Appetite! 
```

File: content/blog/ddd-decameron.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+Another sentence appeared on the screen:
+
+"You could leave, but your team's databases will be dropped and won't be recovered."
+
+Dead silence permeats the room. After a while, somebody finally breaks the ice:
+
+"Well, I'll go first."
+
+## Day 1 From[ donatj](https://news.ycombinator.com/user?id=donatj): Honestly Wish We had a DBA**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:36]:
```suggestion
## Day 1 From [donatj](https://news.ycombinator.com/user?id=donatj): Honestly Wish We had a DBA
```

File: content/blog/ddd-decameron.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+He got let go for basically having a really bad temper, and we decided not to look for a new DBA. I was sad he was gone but also glad to not be getting screamed at. It feels silly, but I've got really unresolved and mixed feelings about the whole thing. I genuinely really liked the guy, loved talking music and movies with him, but he would heel turn in an instant and just ruin your day.
+
+Honestly wish we had a DBA, I ended up picking up a lot of his responsibilities. I'd love to have someone I could learn from. Basically, I'm getting real sick of optimizing reports.
+
+The screen really is sentient, even intelligent at some level. It automatically summarized the story:
+
+"Honestly wish we had a DBA"
+
+## Day 2  From[ steve_adams_86](https://news.ycombinator.com/user?id=steve_adams_86): Both Brilliant, both jerks**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:48]:
```suggestion
## Day 2  From [steve_adams_86](https://news.ycombinator.com/user?id=steve_adams_86): Both Brilliant, both jerks
```

File: content/blog/ddd-decameron.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+One they had a great system in place and the product felt mature, the companies axed them.
+
+The screen falshes:
+
+"All brilliant, all jerks."
+
+DBAs in the room frowned a little. 
+
+## Day 3 From[ jamal-kumar](https://news.ycombinator.com/user?id=jamal-kumar): Gone DBA**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:66]:
```suggestion
## Day 3 From [jamal-kumar](https://news.ycombinator.com/user?id=jamal-kumar): Gone DBA
```

File: content/blog/ddd-decameron.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+I think we called him 'ballgag ***' (fill in the blank with your least favorite person's name) after that
+
+These guys are worth their weight in gold though, I think I heard he ended up getting a job at Microsoft down the line.
+
+The screen switches to today's theme:
+
+"Gone DBA"
+
+## Day 4 From[ Simon_O_Rourke](https://news.ycombinator.com/user?id=Simon_O_Rourke): He was Supposed to Know Everything**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:82]:
```suggestion
## Day 4 From [Simon_O_Rourke](https://news.ycombinator.com/user?id=Simon_O_Rourke): He was Supposed to Know Everything
```

File: content/blog/ddd-decameron.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+Only thing, was this guy had zero programming experience outside SQL and Bash. Anytime he had a question in a meeting about some calling Java process, he used to wait until the meeting was finished and then grab five minutes with me privately to ask about whatever. So there was this perceived weakness that his professional persona "knew everything", which meant he couldn't ask fundamental questions or try to (publicly) learn things outside his domain.
+
+In the end, we moved to Amazon Redshift, and he was progressively moved into a smaller and smaller role, where now he just curates 2-year old JIRA tickets. Probably knows he can't get a similarly well-paying gig elsewhere now.
+
+The familiar flashes appeared once more:
+
+"He was supposed to 'know everything'"
+
+## Day 5 From[ skc](https://news.ycombinator.com/user?id=skc): “UR RUINING MY F****ING DATABASE”**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:94]:
```suggestion
## Day 5 From [skc](https://news.ycombinator.com/user?id=skc): “UR RUINING MY F****ING DATABASE”
```

File: content/blog/ddd-decameron.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+<p id="gdcalert1" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image1.jpg). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert2">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/image1.jpg "image_tooltip")
+
+
+**The DBAs among these 10 people have been humble audiences for 5 days. This picture becomes the last straw to break their nerves. They decided to take the lead now.**
+
+## Day 6 From[ forinti](https://news.ycombinator.com/user?id=forinti): Try to be the person I needed help from before**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:114]:
```suggestion
## Day 6 From [forinti](https://news.ycombinator.com/user?id=forinti): Try to be the person I needed help from before
```

File: content/blog/ddd-decameron.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+And when a bad idea gets implemented, it is really hard to undo it and it tends to just create more chaos: you need materialised views to get around bad modelling, or weird views to compensate for duplicate data, etc.
+
+So, yes, a DBA is a really important role. With regards to it being a good job, it depends if your company takes it seriously and puts you in the development process, otherwise you'll be in for a lot of stress.
+
+Screen displays:
+
+"Try to be the person I needed help from before"
+
+## Day 7 From[ compton93](https://news.ycombinator.com/user?id=compton93): For that company my role was priceless**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:130]:
```suggestion
## Day 7 From [compton93](https://news.ycombinator.com/user?id=compton93): For that company my role was priceless
```

File: content/blog/ddd-decameron.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+
+I worked for several years at a semi-startup as a Database Engineer where my role was to guide developers in writing performant SQL (and writing indexes for them) and architect their schema to be inline with our future plans and various other database tasks (managing query plans, being an expert on the database feature set etc). I even created an internal course: SQL School, that I included the directors/customer support teams in.
+
+To that specific company my role was priceless, new features could be built in 1/2 the time with an expert writing the queries and handling the database. My programmatic analysis of our 20 year old code base had 5000 unique queries and 3000 more when accounting for dynamic SQL. Not gonna lie it was a complete mess but if I didn't exist they would have needed a lot more database resources. It was a terrible code base and any plan to refactor and avoid this mess would still need my role to do the transition.
+
+The screen flashes and shows:
+
+"For that company my role was priceless"
+
+## Day 8 From[ Twisell](https://news.ycombinator.com/user?id=Twisell): Exhausting yet rewarding**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:142]:
```suggestion
## Day 8 From [Twisell](https://news.ycombinator.com/user?id=Twisell): Exhausting yet rewarding
```

File: content/blog/ddd-decameron.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+
+Instead we switched database like 11 times and now I don't know the database either. 
+
+The bugs and poor performance lingers though. 
+
+The screen changed the sentence to:
+
+"Exhausting yet rewarding"
+
+## Day 9 From[ jhugo](https://news.ycombinator.com/user?id=jhugo): Conflict comes from lack of understanding**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:160]:
```suggestion
## Day 9 From [jhugo](https://news.ycombinator.com/user?id=jhugo): Conflict comes from lack of understanding
```

File: content/blog/ddd-decameron.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+
+Networks & DBs present abstractions, but they're very leaky ones, yet developers often expect to just treat them as magical black boxes about which they don't need to know much of the inner workings, constraints, tradeoffs etc. That makes the interface between the network engineer or DBA and software developers a natural site of conflict.
+
+(You even see this above in this very thread, someone complaining that a DBA was punishing developers for "querying the database too much". Most likely the DBA was actually upset about how they were querying the database, not that they were querying it too much, but the developers want it to "just work" so the distinction isn't important to them.) 
+
+Screen flashes:
+
+"Conflict comes from lack of understanding"
+
+## Day 10 From[ niklasrde](https://news.ycombinator.com/user?id=niklasrde) &[ datalopers](https://news.ycombinator.com/user?id=datalopers): The lost art -- data modeling and SQL**
```
Comment (boojack) [inline content/blog/ddd-decameron.md:172]:
```suggestion
## Day 10 From [niklasrde](https://news.ycombinator.com/user?id=niklasrde) &[ datalopers](https://news.ycombinator.com/user?id=datalopers): The lost art -- data modeling and SQL
```

**Review by boojack — CHANGES_REQUESTED** [review]

File: content/blog/ddd-decameron.md [hunk 14]
```diff
@@ -0,0 +1,190 @@
+---
```
Comment (tianzhou) [inline content/blog/ddd-decameron.md:1]: ddd-decameron.md is not SEO friendly.

Let's use "dba-database-developer-decameron"

**Review by tianzhou — CHANGES_REQUESTED** [review]

File: content/blog/ddd-decameron.md [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+title: "DBA, Database and Developer - The Decameron"
```
[the remark this answers is not in the document]
Comment (LucyLu8L) [reply content/blog/ddd-decameron.md:2]: to emphasize our special attention to DBA; plus database is more of a media in these stories, they are more focused on the relationship and interaction between DBAs and Developers

File: content/blog/dba-database-developer-decameron.md [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+## Day 5 From [skc](https://news.ycombinator.com/user?id=skc): “UR RUINING MY F****ING DATABASE”
+
+This is eerily similar to my experiences with two very, very good DBAs I worked with.
+
+I'm now wondering if there is something about the role that lends itself to such behavior. One day they are super pleasant, helpful and friendly. The next they are chewing people out for seemingly minor and random transgressions and saying things like "ruining my fucking database".
+
+Today these guys remind me a lot of JK Simmons character in "Whiplash"
+
+The screen displays an image this time:
+![RoaringDBA](https://bytebase.com/static/blog-changelog-assets/2022/06/roaringdba.jpeg)
```
[the remark this answers is not in the document]
Comment (boojack) [reply content/blog/dba-database-developer-decameron.md:101]: Don't need to add the prefix in `.md` file.

**Review by boojack — CHANGES_REQUESTED** [review]
You can preview the article by https://bytebase-com-git-docs-ddd-decameron-bytebase.vercel.app/blog/dba-database-developer-decameron

**Comment by LucyLu8L** [comment]
all review suggestions committed, please review again

File: content/blog/dba-database-developer-decameron.md [hunk 17]
```diff
@@ -0,0 +1,183 @@
+---
+title: "DBA, Database and Developer - The Decameron"
+author: Lucy
+published_at: 2022/06/06 22:30
+feature_image: /static/blog-changelog-assets/2022/06/roaringdba.jpeg
+tags: Education
+featured: true
+description: This is the first of the DDD series, consisting of 10 stories of DBA & Database & developer.
+---
+A few days ago, we posted a question on HN: [Ask HN: Is DBA still a good job?](https://news.ycombinator.com/item?id=31309729) The question quickly drew the attention of the HN community and got 236 thought-provoking comments. We share and feel the challenges, the frustration, the fatigue and the expectation.
```
Comment (boojack) [inline content/blog/dba-database-developer-decameron.md:10]:
```suggestion

A few days ago, we posted a question on HN: [Ask HN: Is DBA still a good job?](https://news.ycombinator.com/item?id=31309729) The question quickly drew the attention of the HN community and got 236 thought-provoking comments. We share and feel the challenges, the frustration, the fatigue and the expectation. 
```

**Review by boojack — APPROVED** [review]
LGTM

**Review by tianzhou — APPROVED** [review]
LGTM. Ship it

**Pull request merged**